packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -599,7 +599,8 @@ export function getInternalReactConstants(version: string): {
       !shouldSkipForgetCheck &&
       // $FlowFixMe[incompatible-type] fiber.updateQueue is mixed
       (fiber.updateQueue?.memoCache != null ||
-        fiber.memoizedState?.memoizedState?.[REACT_MEMO_CACHE_SENTINEL])
+        (Array.isArray(fiber.memoizedState) &&
+          fiber.memoizedState[0]?.[REACT_MEMO_CACHE_SENTINEL]))
```

Comment:
For backward compat., it should probably also check `fiber.memoizedState?.memoizedState?.[REACT_MEMO_CACHE_SENTINEL]`
```suggestion
        (Array.isArray(fiber.memoizedState) &&
          fiber.memoizedState[0]?.[REACT_MEMO_CACHE_SENTINEL] || fiber.memoizedState?.memoizedState?.[REACT_MEMO_CACHE_SENTINEL]))
```

Reply:
ahh yeah i completely forgot about that even after you had mentioned it 😅